cmd/relayer/main.go

```diff
@@ -0,0 +1,412 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/urfave/cli/v2"
+
+	p2pnet "github.com/athanorlabs/go-p2p-net"
+	rcommon "github.com/athanorlabs/go-relayer/common"
+	rcontracts "github.com/athanorlabs/go-relayer/impls/gsnforwarder"
+	net "github.com/athanorlabs/go-relayer/net"
+	"github.com/athanorlabs/go-relayer/relayer"
+	rrpc "github.com/athanorlabs/go-relayer/rpc"
+
+	"github.com/athanorlabs/atomic-swap/cliutil"
+	"github.com/athanorlabs/atomic-swap/common"
+	contracts "github.com/athanorlabs/atomic-swap/ethereum"
+	swapnet "github.com/athanorlabs/atomic-swap/net"
+
+	logging "github.com/ipfs/go-log"
+)
+
+const (
+	flagDataDir          = "datadir"
+	flagEndpoint         = "endpoint"
+	flagForwarderAddress = "forwarder-address"
+	flagKey              = "key"
+	flagRPC              = "rpc"
+	flagRPCPort          = "rpc-port"
+	flagDeploy           = "deploy"
+	flagLog              = "log-level"
+	// TODO: do we need this, or can we assume all swap relayers will need to be on the p2p network?
+	flagWithNetwork       = "with-network"
+	flagLibp2pKey         = "libp2p-key"
+	flagLibp2pPort        = "libp2p-port"
+	flagBootnodes         = "bootnodes"
+	flagRelayerCommission = "relayer-commission"
+
+	defaultLibp2pPort = 10900
+)
+
+var (
+	log = logging.Logger("main")
+
+	flags = []cli.Flag{
+		&cli.StringFlag{
+			Name:  flagDataDir,
+			Usage: "Path to store swap artifacts",
+			Value: "{HOME}/.atomicswap/{ENV}", // For --help only, actual default replaces variables
+		},
+		&cli.StringFlag{
+			Name:  flagEndpoint,
+			Value: "http://localhost:8545",
+			Usage: "Ethereum RPC endpoint",
+		},
+		&cli.StringFlag{
+			Name:  flagKey,
+			Value: "eth.key",
+			Usage: "Path to file containing Ethereum private key",
+		},
+		&cli.StringFlag{
+			Name:  flagForwarderAddress,
+			Usage: "Address of the forwarder contract to use. Defaults to the forwarder address in the chain's swap contract.",
+		},
+		&cli.UintFlag{
+			Name:  flagRPCPort,
+			Value: 7799,
+			Usage: "Relayer RPC server port",
+		},
+		&cli.BoolFlag{
+			Name:  flagRPC,
+			Value: false,
+			Usage: "Run the relayer RPC server. Defaults false",
```

Comment:
We run the relayer server regardless, if I'm understanding this correctly. I don't think the end user needs to understand that accepting RPC requests is also a "server". Maybe the description should say something like "Accept RPC requests on localhost"? What are the requests that we accept?

Reply:
we don't run the relayer HTTP RPC server if this is set to false. there are now 2 ways for the relayer to accept incoming requests, first is through the p2p network, second is through the RPC server. since users will be discovering the relayers via p2p now (most likely), I don't think it's necessary to have swap relayers also run the RPC server by default. 

the requests it's accepting is only the request to submit a transaction.

Reply:
I'll update the description though